src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/builder/YFlowBuilder.groovy

```diff
@@ -0,0 +1,157 @@
+                    .endpoint(FlowEndpointV2.builder()
+                            .switchId(sw.dpId)
+                            .portNumber(getRandomAvailablePort(sw, topologyDefinition, useTraffgenPorts,
+                                    busyEndpoints.findAll { it.sw == swT.shared.dpId }*.port))
+                            .vlanId(randomVlan(busyEndpoints*.vlan))
+                            .build())
+                    .build()
+            busyEndpoints << new SwitchPortVlan(ep.endpoint.switchId, ep.endpoint.portNumber, ep.endpoint.vlanId)
+            ep
+        }
+        String flowId = generateFlowId()
+        subFlows.first().flowId = "S1." + flowId
+        subFlows.last().flowId = "S2." + flowId
+        this.yFlowRequest = YFlowCreatePayload.builder()
+                .yFlowId(flowId)
+                .sharedEndpoint(se)
+                .subFlows(subFlows)
+                .maximumBandwidth(1000)
+                .ignoreBandwidth(false)
+                .periodicPings(false)
+                .description(generateDescription())
+                .strictBandwidth(false)
+                .encapsulationType(TRANSIT_VLAN.toString())
+                .pathComputationStrategy(COST.toString())
+                .build()
+    }
+
+    YFlowBuilder withProtectedPath(boolean allocateProtectedPath) {
+        this.yFlowRequest.allocateProtectedPath = allocateProtectedPath
+        return this
+    }
+
+    YFlowBuilder withEp1FullPort() {
+        this.yFlowRequest.subFlows.first().endpoint.vlanId = 0
+        return this
+    }
+
+    YFlowBuilder withEp2FullPort() {
+        this.yFlowRequest.subFlows.last().endpoint.vlanId = 0
+        return this
+    }
+
+
+    YFlowBuilder withEncapsulationType(FlowEncapsulationType encapsulationType) {
+        this.yFlowRequest.encapsulationType = encapsulationType.toString()
+        return this
+    }
+
+    YFlowBuilder withEp1QnQ(Integer innerVlan = new Random().nextInt(4095)) {
+        this.yFlowRequest.subFlows.first().endpoint.innerVlanId = innerVlan
+        return this
+    }
+
+    YFlowBuilder withEp2QnQ(Integer innerVlan = new Random().nextInt(4095)) {
+        this.yFlowRequest.subFlows.last().endpoint.innerVlanId = innerVlan
+        return this
+    }
+
+    YFlowBuilder withSharedEpQnQ() {
+        this.yFlowRequest.subFlows.last().sharedEndpoint.innerVlanId = new Random().nextInt(4095)
+        this.yFlowRequest.subFlows.first().sharedEndpoint.innerVlanId = new Random().nextInt(4095)
+        return this
+    }
+
+    YFlowBuilder withEp1AndEp2SameSwitchAndPort() {
+        this.yFlowRequest.subFlows.first().endpoint.switchId = this.yFlowRequest.subFlows.last().endpoint.switchId
+        this.yFlowRequest.subFlows.first().endpoint.portNumber = this.yFlowRequest.subFlows.last().endpoint.portNumber
+        return this
+    }
+
+    /**
+     * Adds Y-Flow and waits for it to become UP.
+     */
+    YFlowExtended create() {
```

Comment:
Let's either rename this method to `.buildAndCreate()` or split builder's finalization method `build()` and flow creation.
Bulder pattern requires such a finalizer named `build()` and it would be easier for others to understand how the code works.